features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -93,17 +95,51 @@ internal class Base64Serializer private constructor(
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
         base64SerializerCallback: Base64SerializerCallback?
     ) {
-        val base64String = convertBmpToBase64(drawable, bitmap, shouldCacheBitmap)
-        finalizeRecordedDataItem(base64String, imageWireframe, base64SerializerCallback)
+        val byteArray = compressBitmap(bitmap)
+        val base64String = convertBmpToBase64(byteArray, drawable, bitmap, shouldCacheBitmap)
+        val hash = calculateHash(drawable, byteArray)
+        finalizeRecordedDataItem(base64String, hash, imageWireframe, base64SerializerCallback)
+    }
+
+    private fun calculateHash(
+        drawable: Drawable,
+        byteArray: ByteArray
+    ): String {
+        val md5Generator = MD5HashGenerator(logger)
```

Comment:
Is usage of `MD5` a requirement by the backend?

Reply:
afaik we can use any hash. MD5 just because it's fast, and is [what is being used on ios](https://github.com/DataDog/dd-sdk-ios/blob/da1f1b99257e36edc114964651fa4a33f7d35d03/DatadogSessionReplay/Sources/Recorder/Utilities/ImageDataProvider.swift#L107)

Reply:
Ok good to know. My point was just that we already have SHA-256 generator and maybe it is worth to share it.

And we are using MD5 for non-cryptographic purpose, so the security alert is obviously a false positive.

Reply:
We could use that, but I'm assuming our primary consideration is speed (and potential collisions) and md5 is almost certainly going to be faster and less computationally intensive